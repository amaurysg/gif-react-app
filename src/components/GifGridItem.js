import React from 'react'
import PropTypes from 'prop-types'

const GifGridItem = ({id, title, url}) => {

  /* console.log(id, title, url) */
  return (
    <div key={id} className="card animate__animated animate__slideInDown" >
      <img src={url} alt={title} />
        <p >{title}</p>
    </div>
  )
}

GifGridItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,

}

export default GifGridItem
