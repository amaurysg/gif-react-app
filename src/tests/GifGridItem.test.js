import React from 'react'
import { shallow } from "enzyme"
import GifGridItem from "../components/GifGridItem"

describe('Test in component GifGridItem.js', ()=>{

  const title = 'Titulo'
  const id = 'Un id'
  const url = 'Un url'
  const wrapper = shallow(<GifGridItem title={title} id={id} url={url} />)
  
  test('El render del componente debe estar Ok', () => {
    

      expect(wrapper).toMatchSnapshot()
      
    })

    test('Debe tener un <p/> con title', () => {
      
      const p = wrapper.find('p').text().trim()
      expect(p).toBe(title)

    })

    test('Debe tener la img = url and alt = title ', () => {

      const img = wrapper.find('img')

      /* Maneras de entrar a html y props */

      /* console.log(img.html()) */
      /* props() para acceder a las propiedas de html */
      /* console.log(img.props()) */
       /*   console.log(img.prop('src'))
         console.log(img.prop('alt')) */

        /* Ahora hacemos el expect */
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)

    })

    test('Debe tener class animate__slideInDown ', () => {
      const div = wrapper.find('div')
      
      expect(div.hasClass('animate__slideInDown')).toBe(true)

      
    })
    
    
    
    




})
