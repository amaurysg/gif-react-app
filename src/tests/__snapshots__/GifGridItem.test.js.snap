// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component GifGridItem.js El render del componente debe estar Ok 1`] = `
<div
  className="card animate__animated animate__slideInDown"
  key="Un id"
>
  <img
    alt="Titulo"
    src="Un url"
  />
  <p>
    Titulo
  </p>
</div>
`;
